R/foreign_keys.R

```diff
@@ -0,0 +1,91 @@
+#' Get FK constraints
+#'
+#' @param con Connection to get constraints for.
+#'
+#' @return A list representing foreign key constraints
+#' @keywords internal
+get_fk_constraints <- function(con) {
+  dialect <- sql_dialect(con)
+  get_fks <- switch(
+    dialect,
+    "sqlite" = get_sqlite_fk,
+    "postgresql" = get_postgres_fk,
+    {
+      stop(sprintf("Only sqlite and postgresql dialects are supported got %s.",
+                   dialect))
+    }
+  )
+  parse_constraints(get_fks(con))
+}
+
+#' Get FK constraints for SQLite connection
+#'
+#' @param con SQLite connection to get constraints for.
+#'
+#' @return A data frame representing foreign key constraints.
+#' @keywords internal
+get_sqlite_fk <- function(con) {
+  tables <- DBI::dbListTables(con)
+
+  constraints <- data.frame(matrix(ncol = 4, nrow = 0),
+                            stringsAsFactors = FALSE)
+  colnames(constraints) <- c("constraint_table", "constraint_column",
+                             "referenced_table", "referenced_column")
+  queries <- lapply(tables, function(table) {
+    sprintf(
+      "SELECT pragma.'table' as referenced_table,
+        pragma.'from' as constraint_column,
+        pragma.'to' as referenced_column,
+        '%s' as constraint_table
+        FROM  pragma_foreign_key_list('%s') as pragma",
+      table, table)
+  })
+
+  constraints <- lapply(queries, function(query) {
+    DBI::dbGetQuery(con, query)
+  })
+  as.data.frame(do.call(rbind, constraints))
```

Comment:
is the `as.data.frame` needed or wanted here? Each of the result of `dbGetQuery` will be a df so rbinding these together will give you a df

Reply:
Not needed, I didn't try to see what would happen without the cast. I've removed it now.